staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -337,5 +345,15 @@ func (s *ServerRunOptions) AddUniversalFlags(fs *pflag.FlagSet) {
 		"This option, if set, represents the maximum amount of grace period the apiserver will wait "+
 		"for active watch request(s) to drain during the graceful server shutdown window.")
 
+	utilfeature.DefaultMutableVersionedFeatureGate.DeferErrorsToValidation(true)
```

Comment:
How can I use a different featuregate instance?

Reply:
This is tied to `utilfeature.DefaultMutableFeatureGate.AddFlag(fs)`. Since the flag is only tied to the DefaultMutableFeatureGate, we cannot use a different featuregate instance

Reply:
> This is tied to `utilfeature.DefaultMutableFeatureGate.AddFlag(fs)`. Since the flag is only tied to the DefaultMutableFeatureGate, we cannot use a different featuregate instance

Can you refactor so that different featuregate instances can be used instead of further pinning an existing one?

Reply:
All the references to the `feature_gate.Enabled` in the api server are referring to the `DefaultFeatureGate`. The instances of feature_gate throughout k8s are fairly inconsistent, with most of them referring to the `DefaultFeatureGate`. I think the refactoring to make different featuregate instances can be used is itself non-trivial and should be in a separate pr, since this is not what this PR is about.